DOC-730: Convert Storage reference to Markdown (#500)
up42/up42-py: docs/reference/storage-reference.md

```python
get_assets(
    created_after,
    created_before,
    workspace_id,
    collection_names,
    producer_names,
    tags,
    sources,
    search,
    limit,
    sortby,
    descending,
    return_json,
)
```

The returned format is `Union[list[Asset], dict]`.

<h5> Arguments </h5>

| Argument           | Overview                                                                                                                                                                                                                                                                                                                                                                                                                                    |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `created_after`    | **Union[str, datetime]**<br/>Search for assets created after the specified timestamp in the `"YYYY-MM-DD"` format.                                                                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| `created_after`    | **Union[str, datetime]**<br/>Search for assets created after the timestamp specified in the `YYYY-MM-DD` format.                                                                                                                                                                                                                                                                                                                         |
```

> | `created_before`   | **Union[str, datetime]**<br/>Search for assets created before the specified timestamp in the `"YYYY-MM-DD"` format.                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| `created_before`   | **Union[str, datetime]**<br/>Search for assets created before the timestamp specified in the `YYYY-MM-DD` format.                                                                                                                                                                                                                                                                                                                         |
```

> | `workspace_id`     | **str**<br/>Search by the workspace ID.                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer: ```suggestion
| `workspace_id`     | **str**<br/>Search by workspace ID.                                                                                                                                                                                                                                                                                                                                                                                                     |
```

| `producer_names`   | **list[str]**<br/>Search for assets from any of the provided [producers](https://docs.up42.com/developers/api-glossary#producers).                                                                                                                                                                                                                                                                                                          |
| `tags`             | **list[str]**<br/>Search for assets with any of the provided tags.                                                                                                                                                                                                                                                                                                                                                                          |

> | `sources`          | **list[str]**<br/>Search for assets from any of the provided sources. The allowed values:<br/><ul><li>`"ARCHIVE"`</li><li>`"TASKING"`</li><li>`"ANALYTICS"`</li><li>`"USER"`</li></ul>                                                                                                                                                                                                                                                      |

Reviewer: Actually, the Analytics team has just told me they dropped `ANALYTICS`

| `search`           | **str**<br/>Search for assets that contain the provided search query in their name, title, or order ID.                                                                                                                                                                                                                                                                                                                                     |
| `limit`            | **int**<br/>The number of assets on a result page.                                                                                                                                                                                                                                                                                                                                                                                          |

> | `sortby`           | **str**<br/>The property to sort by. The allowed values are:<br/><ul><li>`"id"`</li><li>`"workspaceId"`</li><li>`"accountId"`</li><li>`"createdAt"`</li><li>`"updatedAt"`</li><li>`"name"`</li><li>`"size"`</li><li>`"contentType"`</li><li>`"geospatialMetadataExtractionStatus"`</li><li>`"productId"`</li><li>`"orderId"`</li><li>`"producerName"`</li><li>`"collectionName"`</li><li>`"source"`</li>The default value is `"createdAt"`. |

Reviewer: For sortby, descending, and return_json, please reuse the text from https://sdk.up42.com/reference/project-reference/#get_jobs

Reviewer: Sortby values will be different here + please remove `"`

Reviewer: `</li>The default value is "createdAt". |`
You need to close the list with `</ul>`

| `return_json`      | **bool**<br/>Determines how to return the assets.<br/><ul><li>`True`: returns a JSON dictionary</li><li>`False`: returns a list of asset objects.</li></ul>The default value is `False`.                                                                                                                                                                                                                                                    |

<h5> Example </h5>

```python
storage.get_assets(
    created_after="2021-01-01",
    created_before="2023-01-01",
    workspace_id="68567134-27ad-7bd7-4b65-d61adb11fc78",
    collection_names=["phr"],
    producer_names=["airbus"],
    tags=["optical", "us"],
    sources=["ARCHIVE"],
    search=["NY Central Park"],
    limit=5,
    sortby="productId",
    return_json=True,
)
```

## Orders

> ### get_orders()

Reviewer: Let's also support [this release](https://github.com/up42/up42-py/pull/465/files#diff-f9cb3611528e0a7c83ebe23981dd51766dab5af53b9d6396e714552008d75f41R314) here:
- New arguments: 
  - `workspace_orders: bool = True`
  - `status: Optional[List[str]] = None`
  - `name: Optional[str] = None`
- Status values: 
  - ` "CREATED",
            "BEING_PLACED",
            "PLACED",
            "PLACEMENT_FAILED",
            "DELIVERY_INITIALIZATION_FAILED",
            "BEING_FULFILLED",
            "DOWNLOAD_FAILED",
            "DOWNLOADED",
            "FULFILLED",
            "FAILED_PERMANENTLY"` (+ a link to the [description](https://docs.up42.com/developers/api-tasking/tasking-monitor#order-statuses))

> The `get_orders()` function allows you to search for tasking and catalog orders in the workspace.

Reviewer: ```suggestion
The `get_orders()` function allows you to search for tasking and catalog orders.
```

```python
get_orders(
    return_json,
    limit,
    sortby,
    descending,
    order_type,
    tags,
)
```

The returned format is `Union[list[Order], dict]`.

<h5> Arguments </h5>

| Argument      | Overview                                                                                                                                                                                                                                                |
| ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `return_json` | **bool**<br/>Determines how to return the orders.<br/><ul><li>`True`: returns a JSON dictionary</li><li>`False`: returns a list of order objects.</li></ul>The default value is `False`.                                                                |

> | `limit`       | **int**<br/>The number of assets in an order to return.                                                                                                                                                                                                 |

Reviewer: Please make it consistent with the other `limit` description.

> | `sortby`      | **str**<br/>The results sorting method that arranges the returned orders. The allowed values:<br/><ul><li>`"createdAt"`</li><li>`"updatedAt"`</li><li>`"status"`</li><li>`"dataProvider"`</li><li>`"type"`</li></ul>The default value is `"createdAt"`. |

Reviewer: The same as above

> | `descending`  | **bool**<br/>Determines the sorting order of the returned orders.<br/><ul><li>`True`: descending</li><li>`False`: ascending</li></ul>The default value is `True`.                                                                                       |

Reviewer: The same as above

> | `order_type`  | **str**<br/>The type of orders to return. The allowed values:<br/><ul><li>`"TASKING"`: use to get only tasking orders</li><li>`"ARCHIVE"`: use to get only catalog orders</li></ul>Don't use the parameter to get orders of all types.                  |

Reviewer: Remove `"`

Reviewer: "Don't use the parameter to get orders of all types." (I see that you took it from the reference, but I think there's a better way to describe it) -> To get orders of all types, omit the parameter.

Reviewer: There's something wrong with line spacing above the sentence

Author: After much investigation, I think the line spacing is like this because the sentence `To get orders of all types, omit the parameter.` doesn't have any text within the back ticks, like how the others have the default value written `True`. 
In my commit, I can add an empty ` ` at the end of the sentence, and it will add the line spacing. But moving this sentence to the top of the description or omitting the sentence entirely would solve the problem, too. 
What is your preference? @dustgalactic

Reviewer: Let's move it to the top :)

| `return_json`      | **bool**<br/>Determines how to return assets.<br/><ul><li>`True`: return JSON.</li><li>`False`: return a list of asset objects.</li></ul>The default value is `False`.                                                                                                                                                                                                                                                                                                 |

<h5> Example </h5>

```python
storage.get_assets(
    created_after="2021-01-01",
    created_before="2023-01-01",
    workspace_id="68567134-27ad-7bd7-4b65-d61adb11fc78",
    collection_names=["phr"],
    producer_names=["airbus"],
    tags=["optical", "us"],
    sources=["ARCHIVE"],
    search=["NY Central Park"],
    limit=5,
    sortby="productId",

>     return_json=True,

Reviewer: Let's include `descending` in the example.

## Orders

### get_orders()

The `get_orders()` function allows you to search for tasking and catalog orders.

```python
get_orders(
    workspace_orders
    return_json,
    limit,
    sortby,
    descending,
    order_type,
    status,
    name,
    tags,
)
```

The returned format is `Union[list[Order], dict]`.

<h5> Arguments </h5>

| Argument           | Overview                                                                                                                                                                                                                                                                                                                                                                                                            |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `workspace_orders` | **bool**<br/>Determines the orders to return.<br/><ul><li>`True`: return workspace orders.</li><li>`False`: return all account orders.</li></ul>The default value is `True`.                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| `workspace_orders` | **bool**<br/>Determines the scope of orders to return.<br/><ul><li>`True`: return orders from your workspace.</li><li>`False`: return orders from the entire account.</li></ul>The default value is `True`.                                                                                                                                                                                                                                        |
```

| `return_json`      | **bool**<br/>Determines how to return orders.<br/><ul><li>`True`: return JSON.</li><li>`False`: return a list of order objects.</li></ul>The default value is `False`.                                                                                                                                                                                                                                              |

> | `limit`            | **int**<br/>The number of assets on a result page.                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| `limit`            | **int**<br/>The number of orders on a result page.                                                                                                                                                                                                                                                                                                                                                                  |
```

| `status`           | **list[str]**<br/>The [status](https://docs.up42.com/developers/api-tasking/tasking-monitor#order-statuses) of the order. The allowed values:<br/><ul><li>`CREATED`</li><li>`BEING_PLACED`</li><li>`PLACED`</li><li>`PLACEMENT_FAILED`</li><li>`DELIVERY_INITIALIZATION_FAILED`</li><li>`BEING_FULFILLED`</li><li>`DOWNLOAD_FAILED`</li><li>`DOWNLOADED`</li><li>`FULFILLED`</li><li>`FAILED_PERMANENTLY`</li></ul> |
| `name`             | **str**<br/>The name of the order.                                                                                                                                                                                                                                                                                                                                                                                  |

> | `tags`             | **list[str]**<br/>Search for orders with any of the provided tags.                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| `tags`             | **list[str]**<br/>Order tags. Use to search for orders with any of the provided tags.                                                                                                                                                                                                                                                                                                                                                  |
```

<h5> Example </h5>

```python
storage.get_orders(
    workspace_orders=False,
    return_json=True,
    limit=2,
    sortby="status",

>     status="FULFILLED",

Reviewer: Let's stick to the order in the table above: `order_type` first, then `status`.
